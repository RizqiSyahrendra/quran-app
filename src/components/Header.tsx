"use client";

export default function Header() {
    return (
        <div className="block md:hidden px-5 py-4 text-xl text-header-font font-bold">
            Qur'an App
        </div>
    );
}