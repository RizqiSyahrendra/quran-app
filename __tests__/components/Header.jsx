import { render } from '@testing-library/react';
import Header from '@/components/Header';
import '@testing-library/jest-dom';

// Mock useRouter:
jest.mock("next/navigation", () => ({
    useRouter() {
        return {
            prefetch: () => null,
            push: () => null,
            back: () => null,
        };
    }
}));

describe('Header', () => {

    it('render a Header', () => {
        const { getByText } = render(
            <Header 
                title="Judul"
            />
        );
        const text = getByText("Judul");
        expect(text).toBeInTheDocument();
    });

    it('render Header unchanged', () => {
        const { container } = render(<Header />);
        expect(container).toMatchSnapshot();
    });

});