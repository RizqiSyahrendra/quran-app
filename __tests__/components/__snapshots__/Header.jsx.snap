// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header render Header unchanged 1`] = `
<div>
  <div
    class="flex pt-2 pb-3 mb-4 text-xl text-header-font font-bold "
  >
    <img
      alt="back"
      class="self-center mr-4 cursor-pointer"
      data-nimg="1"
      decoding="async"
      height="24"
      loading="lazy"
      src="/_next/image?url=%2Fassets%2Ficons%2Fback.png&w=48&q=100"
      srcset="/_next/image?url=%2Fassets%2Ficons%2Fback.png&w=32&q=100 1x, /_next/image?url=%2Fassets%2Ficons%2Fback.png&w=48&q=100 2x"
      style="color: transparent;"
      width="24"
    />
    <div />
  </div>
</div>
`;
